ConditionalTranslation.md

```diff
@@ -0,0 +1,297 @@
+# Conditional Translation
+
+## Overview
+
+> This section is non-normative
+
+Conditional translation is a mechanism to modify the output source code based on parameters passed to the *WESL translator*.
+This specificaton extends the [*attribute* syntax](https://www.w3.org/TR/WGSL/#attributes) with a new `@if` attribute.
+This attribute indicates that the syntax node it decorates can be removed by the *WESL translator* based on feature flags.
+
+> NOTE: This implementation is similar to the `#[cfg(feature = "")]` syntax in rust.
+
+### Usage Example
+
+```rs
+// global variables and bindings...
+@if(textured)
+@group(0) @binding(0) var my_textuer: texture_2d<f32>;
+
+// structs declarations and struct members...
+struct Ray {
+  position: vec4f,
+  direction: vec4f,
+  @if(debug_mode && raytracing_enabled)
+  ray_steps: u32,
+}
+
+// function declarations, parameters and statements...
+fn main() -> vec4 {
+  @if(legacy_implementation || (is_web_version && xyz_not_supported))
+  let result = legacy_impl();
+  @if(!legacy_implementation && !(is_web_version && xyz_not_supported))
+  let result = modern_impl();
+}
+```
+
+## Definitions
+
+* **Translate-time expression**: A *translate-time expression* is evaluated by the WESL translator and eliminated after the translation. It lives in the *translate-time scope*.
+  Its grammar is a subset of normal WGSL [expressions](https://www.w3.org/TR/WGSL/#expressions). it must be one of:
+  * a *translate-time feature*,
+  * a [logical expression](https://www.w3.org/TR/WGSL/#logical-expr): logical not (`!`), short-circuiting AND (`&&`), short-circuiting OR (`||`),
+  * a [parenthesized expression](https://www.w3.org/TR/WGSL/#parenthesized-expressions),
+  * a boolean literal value (`true` or  `false`).
+
+* **Translate-time scope**: The *translate-time scope* this is an independent scope from the [*module scope*](https://www.w3.org/TR/WGSL/#module-scope), meaning it cannot see any declarations from the source code, and its identifers are independent.
+
+* **Translate-time feature**: A *translate-time feature* is an identifier that evaluates to a boolean. It is set to `true` if the feature is *enabled* during the translation phase and `false` if the feature is *disabled*.
+
+* **Translate-time attribute**: A *translate-time attribute* is parametrized by a *translate-time expression*. It is eliminated after the translation but can affect the syntax node it decorates.
+
+## Location of *Translate-time attributes*
+
+A *translate-time attribute* can appear in all places where the WGSL grammar allows attributes, except if removal of the decorated syntax node would lead to syntactically incorrect code.
+The grammar is extended to allow attributes in several locations previously not allowed by the WGSL grammar. These extensions are specified in section [updated grammar](#updated-grammar).
```

Comment:
can you summarize the attributable locations for the programmer to understand? The precise details are important to specify, but it's a long list to expect the programmer to remember.

Reply:
:+1: can do